### PR TITLE
arcconf device specific tests - v1

### PR DESCRIPTION
**Pull request by manvanthar**
arcconf_drive_oper.py tests arcconf tools device specific configs.

Signed-off-by: Manvanthara B Puttashankar <manvanth@linux.vnet.ibm.com>

[results.txt](https://github.com/avocado-framework-tests/avocado-misc-tests/files/627039/results.txt)
[job.txt](https://github.com/avocado-framework-tests/avocado-misc-tests/files/627040/job.txt)

### CODE REVIEW

**Comment by PraveenPenguin** [comment]
@narasimhan-v can you please help to review this patch

File: io/disk/arcconf/arcconf_drive_oper.py [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+                                                  self.option_args,
+                                                  disk_val,
+                                                  self.option_args2)
+                    self.check_pass(cmd, "Failed to run %s" % cmd)
+
+    def cmdop_list(self, cmd):
+        """
+        Function returns the output of a command
+        """
+        val = process.run(cmd, shell=True, allow_output_check='stdout')
```
Comment (narasimhan-v) [inline io/disk/arcconf/arcconf_drive_oper.py]: Guess `ignore_status=True` is needed here.

File: io/disk/arcconf/arcconf_drive_oper.py [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+              | sed -e "s/\]//g" | sed -e "s/\[//g" | tr \'\\n\' \' \' \
+              | awk \'{print $2,$3,$4,$5}\') ; done'
+        pci_id_formatted = []
+        for i in self.cmdop_list(cmd).splitlines():
+            pci_id_formatted.append(str(i.replace(" ", ":")))
+
+        # check if all the yaml parameters are entered
+        if self.crtl_no is '' or self.channel_no is '' or self.disk_no is \
+           '' or self.pci_id is '' or len(self.disk_no) <= 1 or \
+           self.tool_name is '' or self.http_path is '':
```
Comment (narasimhan-v) [inline io/disk/arcconf/arcconf_drive_oper.py]: `len(self.disk_no)` is redundant here.

File: io/disk/arcconf/arcconf_drive_oper.py [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+                    if "setphy" in self.option:
+                        cmd = "%s %s %s" % (self.option, self.crtl_no,
+                                            self.option_args)
+                    else:
+                        cmd = "%s %s %s %s %s" % (self.option,
+                                                  self.crtl_no,
+                                                  self.option_args,
+                                                  disk_val,
+                                                  self.option_args2)
+                    self.check_pass(cmd, "Failed to run %s" % cmd)
```
Comment (narasimhan-v) [inline io/disk/arcconf/arcconf_drive_oper.py]: `self.check_pass(cmd, "Failed to run %s" % cmd)` in lines 134 and 147 can be made into one.

File: io/disk/arcconf/arcconf_drive_oper.py (hunk 2 above)
Comment (manvanthar) [inline io/disk/arcconf/arcconf_drive_oper.py]: @narasimhan-v , its on purpose.    the first one looks for any empty string and the second one looks for the miminal disk count.

File: io/disk/arcconf/arcconf_drive_oper.py.data/Readme [hunk 4]
```diff
@@ -0,0 +1,10 @@
+arcconf - Command line interface to configure PMC-Sierra
+(Microsemi) Controllers.
+
+arcconf_drive_oper.py covers arcconf functionality 
+specific to physical and logical drive settings.
+ 
+Note:
+    1. supported controllers pci_ids can be entered 
+       seperated by ",".
+    2. TODO: IMAGEUPDATE, SETPOWER, UNINIT
```
Comment (manvanthar) [inline io/disk/arcconf/arcconf_drive_oper.py.data/Readme]: sure @narasimhan-v will do.

Comment (manvanthar) [inline io/disk/arcconf/arcconf_drive_oper.py.data/Readme]: @narasimhan-v , code changed to 

Note:
    1. supported controllers pci_ids can be entered
       seperated by ",".
    2. TODO: IMAGEUPDATE, SETPOWER, UNINIT
    3. Irrepsective of number of disks specified at yaml, 
       Script picks only the first two disks for test 
       optimization.

File: io/disk/arcconf/arcconf_drive_oper.py (hunk 1 above)
Comment (manvanthar) [inline io/disk/arcconf/arcconf_drive_oper.py]: Done. the fix is in v2.

**Comment by manvanthar** [comment]
@PraveenPenguin @narasimhan-v ,   if no further  comments will respin the code. please confirm

**Comment by PraveenPenguin** [comment]
@narasimhan-v can you please comment

**Comment by narasimhan-v** [comment]
@manvanthar no further comments from me.

**Comment by manvanthar** [comment]
thanks.   respinning the v2

**Pull request closed without merging**